python/foxglove-sdk/python/foxglove/_foxglove_py/__init__.pyi

```diff
@@ -127,9 +137,13 @@ def shutdown() -> None:
     """
     ...
 
-def record_file(path: str) -> MCAPWriter:
+def open_mcap(path: str | Path, allow_overwrite: bool = False) -> MCAPWriter:
```

Comment:
Thinking more about your point that `open()` in python can mean read or write - I could imagine a world where we might want to expose mcap read interfaces (e.g. open an existing mcap file and append some additional foxglove-schema channels).

So one idea would be to name this `write_mcap()`.

I'm not sure I like it yet - just throwing the idea out there.